sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java

```diff
@@ -32,8 +32,8 @@ protected AbstractQueryGenerator() {
     }
 
     private String generateQueryParameter(@NonNull String subject) {
-        // user.name is not valid sql parameter identifier.
-        return subject.replaceAll("\\.", "_") + UUID.randomUUID().toString().replaceAll("-", "_");
+        // user.name, user['name'] or user["first name"] are not valid sql parameter identifiers.
+        return subject.replaceAll("[^a-zA-Z\\d]", "_") + UUID.randomUUID().toString().replaceAll("-", "_");
```

Comment:
Why are we replacing the extra character d at the end ?

Reply:
@kushagraThapar actually this results to `\d` only because of the way the Java compiler works (the compiler will replace `\\` with `\`), so after the compiler substitutes the two backslashes with a single one,`\d` would be interpreted by the regex engine as any digit (0-9). 

This renders the whole expression as anything that's not in the class of characters `a-z`, `A-Z` or `0-9`. So any non-alphanumeric character will be converted to a `_`.